src/test/java/seedu/address/logic/commands/FindCommandTest.java

```diff
@@ -68,24 +79,116 @@ public void execute_zeroKeywords_noPersonFound() {
+        assertEquals(expectedModelWithRental.getFilteredPersonList(), modelWithRental.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_tenantsKeyword_success() {
+        String address = "carl";
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1);
+
+        NameContainsKeywordsPredicate namePredicate = prepareNamePredicate(address);
+        PhoneContainsKeywordsPredicate phonePredicate = preparePhonePredicate(address);
+        EmailContainsKeywordsPredicate emailPredicate = prepareEmailPredicate(address);
+        RentalInformationContainsKeywordsPredicate rentalInfoPredicate = prepareRentalInfoPredicate(address);
+
+        FindCommand command = new FindCommand(namePredicate, phonePredicate, emailPredicate, rentalInfoPredicate);
+        expectedModelWithRental.updateFilteredPersonList(
+                namePredicate.or(phonePredicate).or(emailPredicate).or(rentalInfoPredicate));
+        assertCommandSuccess(command, modelWithRental, expectedMessage, expectedModelWithRental);
+        assertEquals(expectedModelWithRental.getFilteredPersonList(), modelWithRental.getFilteredPersonList());
+    }
 
-    //    @Test
-    //    public void execute_multipleKeywords_multiplePersonsFound() {
-    //        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
-    //        NameContainsKeywordsPredicate predicate = preparePredicate("Kurz Elle Kunz");
-    //        FindCommand command = new FindCommand(predicate);
-    //        expectedModel.updateFilteredPersonList(predicate);
-    //        assertCommandSuccess(command, model, expectedMessage, expectedModel);
-    //        assertEquals(Arrays.asList(CARL, ELLE, FIONA), model.getFilteredPersonList());
-    //    }
+    @Test
+    public void execute_rentalStartDateKeyword_success() {
+        String address = "01 Jan";
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+
+        NameContainsKeywordsPredicate namePredicate = prepareNamePredicate(address);
+        PhoneContainsKeywordsPredicate phonePredicate = preparePhonePredicate(address);
+        EmailContainsKeywordsPredicate emailPredicate = prepareEmailPredicate(address);
+        RentalInformationContainsKeywordsPredicate rentalInfoPredicate = prepareRentalInfoPredicate(address);
+
+        FindCommand command = new FindCommand(namePredicate, phonePredicate, emailPredicate, rentalInfoPredicate);
+        expectedModelWithRental.updateFilteredPersonList(
+                namePredicate.or(phonePredicate).or(emailPredicate).or(rentalInfoPredicate));
+        assertCommandSuccess(command, modelWithRental, expectedMessage, expectedModelWithRental);
+        assertEquals(expectedModelWithRental.getFilteredPersonList(), modelWithRental.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_rentalEndDateKeyword_success() {
+        String address = "31 Dec";
```

Comment:
Same as above, should consider use rentalEndDate as variable name.

Reply:
Thanks for pointing that out!